gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/artifacts/DependencyDiscovery.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.artifacts;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.ArtifactView;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.transform.TransformAction;
+import org.gradle.api.artifacts.transform.TransformParameters;
+import org.gradle.api.artifacts.type.ArtifactTypeDefinition;
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.attributes.Category;
+import org.gradle.util.GUtil;
+
+public final class DependencyDiscovery {
+    public static final Attribute<String> ARTIFACT_FORMAT = Attribute.of("artifactType", String.class);
+    public static final String PRODUCT_DEPENDENCIES = "product-dependencies";
+
+    public static ArtifactView getFilteredArtifact(Project project, Configuration conf, String targetArtifact) {
+        return conf.getIncoming().artifactView(v -> {
+            v.getAttributes().attribute(ARTIFACT_FORMAT, targetArtifact);
+            Category projectCategory =
+                    project.getObjects().named(Category.class, PreferProjectCompatabilityRule.PROJECT);
+            v.getAttributes().attribute(Category.CATEGORY_ATTRIBUTE, projectCategory);
+        });
+    }
+
+    public static Configuration copyClasspath(Project project, String name) {
+        Configuration conf = project.getConfigurations()
+                .create(GUtil.toLowerCamelCase("runtimeClasspathFor " + name), c -> {
```

Comment:
Is it worth making this configuration name a bit more specific to sls-packaging/productdeps? I could just imagine someone else making one with the same name in a different plugin.